### PR TITLE
chore: CodeQL fixes, regex refactoring

### PR DESCRIPTION
**Pull request by Jumas**
### Proposed changes

Describe the use case and detail of the change. If this PR addresses an issue on GitHub, make sure to include a link to that issue using one of the [supported keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) here in this description (not in the title of the PR).

### Checklist

Before creating a PR, run through this checklist and mark each as complete:
- [ ] I have read the [`CONTRIBUTING`](CONTRIBUTING.md) document
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] I have updated any relevant documentation

### CODE REVIEW

**Review by yurtsevich-sbb — APPROVED** [review]

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <component name="ProjectCodeStyleConfiguration">
   <code_scheme name="Project" version="173">
+    <option name="LINE_SEPARATOR" value="&#10;" />
```
Comment (grigoriev) [inline .idea/codeStyles/Project.xml:3]: Do we need this setting for idea project?

File: pom.xml [hunk 2]
```diff
@@ -51,6 +51,7 @@
         <testcontainers.version>1.19.8</testcontainers.version>
         <javax.transaction-api.version>1.3</javax.transaction-api.version>
         <byte-buddy.version>1.14.17</byte-buddy.version>
+        <re2j.version>1.7</re2j.version>
```
Comment (grigoriev) [inline pom.xml:54]: could you please move it to  pdfbox and okapi.xliff dependencies and comment these deps as  ?

File: .idea/codeStyles/Project.xml (hunk 1 above)
Comment (Jumas) [reply .idea/codeStyles/Project.xml:3]: Removed.

File: pom.xml (hunk 2 above)
Comment (Jumas) [reply pom.xml]: Done. Please check whether it is what you've expected.

**Review by grigoriev — APPROVED** [review]

**Pull request merged**